### PR TITLE
Setup github workflow for CI

### PR DESCRIPTION
**Pull request by davide125**
Basic config to replicate what we currently have in CircleCI.

### CODE REVIEW

**Comment by davide125** [comment]
Note: I made this on a branch of the repo (as opposed to a fork) because otherwise there's no way to test it. Once it's merged future changes will work as normal using regular PRs.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        ruby: [2.4, 2.5, 2.6]
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: ${{ matrix.ruby }}
+    - name: Install dependencies
```
Comment (jaymzh) [inline .github/workflows/ci.yml:17]: I'm vaguely surprised you don't need to force cmake to be installed, but this looks good.

**Comment by jaymzh** [comment]
example:
https://github.com/jaymzh/sugarjar/blob/master/.github/workflows/lint.yml#L25

**Comment by jaymzh** [comment]
Also mdl if you want to for readmes (https://github.com/jaymzh/sugarjar/blob/master/.github/workflows/lint.yml#L30)

**Comment by davide125** [comment]
Oh neat, I didn't know the rubocop action was a thing. I'll give it a try.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: Continuous Integration
+on: [push, pull_request]
```
Comment (jaymzh) [inline .github/workflows/ci.yml:2]: you probably don't want this, you'll double-test everything. You probably want:
```
on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]
```

**Comment by davide125** [comment]
I can't seem to be able to get custom actions to work at all -- the rubocop one fails because it can't find rubocop, and mdl fails starting the container.

**Comment by jaymzh** [comment]
push your code with the rubocop action so I can poke at it?

**Comment by davide125** [comment]
Done. Also got mdl working by using 1.2.0 instead of 1.2.1

**Pull request merged**